alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/ProcessorAttributeService.java

```diff
@@ -0,0 +1,85 @@
+package eu.xenit.alfresco.processor.service;
+
+import lombok.AllArgsConstructor;
+import org.alfresco.service.cmr.attributes.AttributeService;
+import org.alfresco.service.cmr.attributes.DuplicateAttributeException;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.util.ParameterCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+/**
+ * {@link AttributeService} wrapper that allows different Alfresco nodes to claim ranges of transactions.
+ */
+public class ProcessorAttributeService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorAttributeService.class);
+
+    public static final String ATTR_KEY_HEALTH_PROCESSOR = "HealthProcessor";
+    public static final String ATTR_KEY_IS_RUNNING = "IsRunning";
+
+    protected final AttributeService attributeService;
+    protected final DescriptorService descriptorService;
+
+    public <T extends Serializable> T getAttribute(final String key, final T defaultValue) {
+        final List<T> returnValues = new ArrayList<>();
+
+        if(!attributeService.exists(ATTR_KEY_HEALTH_PROCESSOR,key)) {
+            return defaultValue;
+        }
+
+        attributeService.getAttributes((Long id, Serializable value, Serializable[] keys) -> {
+            logger.debug("Processing attribute id: '{}', keys: '{}', value: '{}'", id, keys, value);
+            returnValues.add(extractTransactionStartId(key, keys));
```

Comment:
I'm a bit confused. 

The method `getAttribute(final String key, final T defaultValue)` implies that it can be used to just get an attribute. 
However the implementation uses `extractTransactionStartId`. 

I think we should make a wrapper around the `AttributeService` that either only uses the "health-processor" conceptual tems, or uses non of them.

Reply:
Renamed to the more generic `extractAttribureValue`